qiita_db/portal.py

```diff
@@ -5,20 +5,145 @@
-        """Returns study id for all Studies belonging to a portal
+
+class Portal(QiitaObject):
+    _table = 'portal_type'
+
+    def __init__(self, portal):
+        self.portal = portal
+        portal_id = convert_to_id(portal, 'portal_type', 'portal')
+        super(Portal, self).__init__(portal_id)
+
+    @classmethod
+    def create(cls, portal, desc):
+        """Creates a new portal on the system
+
+        Parameters
+        ----------
+        portal : str
+            The name of the portal to add
+        desc : str
+            Description of the portal
+
+        Raises
+        ------
+        QiitaDBDuplicateError
+            Portal already exists
+        """
+        if cls.exists(portal):
+            raise QiitaDBDuplicateError("Portal", portal)
+
+        # Add portal and default analyses for all users
+        sql = """DO $do$
+            DECLARE
+                pid bigint;
+                eml varchar;
+                aid bigint;
+            BEGIN
+                INSERT INTO qiita.portal_type (portal, portal_description)
+                VALUES (%s, %s) RETURNING portal_type_id INTO pid;
+
+                FOR eml IN
+                    SELECT email FROM qiita.qiita_user
+                LOOP
+                    INSERT INTO qiita.analysis
+                    (email, name, description, dflt, analysis_status_id)
+                    VALUES (eml, eml || '-dflt', 'dflt', true, 1) RETURNING
+                    analysis_id INTO aid;
+
+                    INSERT INTO qiita.analysis_workflow (analysis_id, step)
+                    VALUES (aid, 2);
+
+                    INSERT INTO qiita.analysis_portal
+                    (analysis_id, portal_type_id)
+                    VALUES (aid, pid);
+                END LOOP;
+            END $do$;"""
+        conn_handler = SQLConnectionHandler()
+        conn_handler.execute(sql, [portal, desc])
+
+        return cls(portal)
+
+    @staticmethod
+    def delete(portal):
+        """Removes a portal from the system
 
         Parameters
         ----------
         portal : str
-           Portal to check studies belong to
+            The name of the portal to add
+        desc : str
+            Description of the portal
+
+        Raises
+        ------
+        QiitaDBError
+            Portal has analyses or studies attached to it
+        """
+        conn_handler = SQLConnectionHandler()
+        # Check if attached to any studies
+        portal_id = convert_to_id(portal, 'portal_type', 'portal')
+        sql = """SELECT study_id from qiita.study_portal
+                 WHERE portal_type_id = %s"""
+        studies = conn_handler.execute_fetchall(sql, [portal_id])
+        if studies:
+            raise QiitaDBError(
+                "Studies still attached to portal %s: %s" %
+                (portal, ', '.join([str(s[0]) for s in studies])))
+
+        # Check if attached to any analyses
+        sql = """SELECT analysis_id from qiita.analysis_portal
```

Comment:
This will always returns some analyses given that the default analyses are attached to the portal right? You will need to modify this line to avoid counting those analyses.